docs/conceptual/core/consensus.md

```diff
@@ -2,14 +2,104 @@
+
+DIAGRAM
+
+### GHOST Rule
+The Greedy Heaviest Observed SubTree rule says that, starting at the genesis block, each fork is resolved by choosing the branch that has the most blocks built on it recursively.
+
+DIAGRAM
+
+## Block Production
+Some nodes in a blockchain network are able to produce new blocks. Exactly which nodes may author blocks depends on which consensus engine you're using. In a centralized network, a single node might author all the blocks, whereas in a completely permissionless network, any node that wishes to, may produce a block.
+
+Nodes that are following the rules of the consensus protocol always produce new blocks on top of the chain that the fork choice rule tells them to. But blockchains still must tolerate nodes that cheat and build blocks in the wrong place. Considering the example fork choice rules we studied above, it's easy to see that an attacker who wanted to revert many blocks worth of transactions could do so by building a longer chain.
+
+To prevent malicious nodes from creating attack chains and overwhelming the honest chain whenever they please, block production must be throttled so that nodes can only produce blocks at a given rate. There are two common ways of achieving this throttling.
+
+### Proof of Work
+In proof of work systems like Bitcoin, any node may produce a block at any time, so long as they have solved a computationally-intensive problem. Solving the problem takes cpu time, and thus miners can only produce blocks in proportion with their computing resources. Substrate provides a Proof of Work consensus engine.
+
+### Slots
+When using a slot based consensus algorithm, there must be a known set of validators who are permitted to produce blocks. Time is divided up into discrete slots, and during each slot only some of the validators may produce a block. The specifics of which validators can author blocks during each slot vary from engine to engine. Substrate provides Aura and Babe, both of which are slot-based block authoring engines.
+
+## Finality
+Transactors in any system want to know when their transaction is finalized, and blockchain is no different. In some traditional systems, finality happens when a receipt is handed over, or papers are signed.
+
+Using the block authoring schemes and fork choice rules described so far, transactions are never truly finalized with certainty. There is always a chance that a longer (or heavier) chain will come along and revert your transaction. However, the more blocks are built on top of a particular block, the less likely it is to ever be reverted. In this way block authoring along with a proper fork choice rule provide probabilistic finality.
+
+When true finality is desired, an additional game, known as a finality gadget can be added to the blockchain's logic. In such a game, members of a fixed authority set cast finality votes, and when a certain threshold is reached, the block is deemed finalized, and will never be reverted.
```

Comment:
is that threshold >2/3n + 1 for n validators? should it be mentioned -- when I read "certain" threshold, I immediately wonder what the required threshold for finality is

Reply:
I had the same thought. the reason I didn't write it in yet, is because I wasn't sure whether there were other finality gadgets (possibly with weaker guarantees) that have a lower threshold. 2n/3 + 1 is definitely common and afaik ideal, but I can imagine inventing a finality gadget that requires 100% voting threshold.

Maybe it's worth mentioning that 2/3 is the most common and linking a BFT paper in the learn more section?

Reply:
Yes, would be nice to link the required lower bound for BFT finality.
```suggestion
When true finality is desired, an additional game, known as a finality gadget can be added to the blockchain's logic. In such a game, members of a fixed authority set cast finality votes for specific blocks. The threshold for achieving [Byzantine Fault Tolerance](https://people.eecs.berkeley.edu/~luca/cs174/byzantine.pdf) requires that greater than 2/3rds of the validator set's voting power commits to each finalized block. This quorum requirement for finality could also be configured to require more support for greater guarantees against reversion of finalized blocks.
```

Reply:
Also, I don't want to be *that* pedantic, but it's not really "true finality". @joepetrowski gave a great presentation on this a while back, I wonder what he thinks of this part

Reply:
There are a few problems here. I'm wary of adjectives and adverbs in general, especially ones like "true" and "always." There are indeed other protocols that require higher thresholds like 4/5. There is a proof that you can never do _better_ than 2/3, though. A higher threshold does not in any way provide a better guarantee (and therein you see the problem with the adjective "_true_ finality," you can't have something be "true" if its guarantees are on a sliding scale (and are they even guarantees then?)).

You should say "deterministic finality" and emphasize that it provides safety. You cannot re-org a finalized block without a hard fork - which requires some external coordination. The chain cannot simply re-org below the finalized head the way a chain with probabilistic finality can.

Reply:
@4meta5 I appreciate it if you can share the link of the presentation.